models/vrp.rb

```diff
@@ -138,93 +138,8 @@ def self.create(hash, delete = true)
-        }
-      }
-
-      configuration = hash[:configuration]
-      return unless configuration
-
-      # configuration consistency
-      if configuration[:preprocessing]
-        if configuration[:preprocessing][:partitions]&.any?{ |partition| partition[:entity].to_sym == :work_day }
-          if hash[:services].any?{ |s|
-              if hash[:configuration][:schedule][:range_indices][:end] <= 6
-                (s[:visits_number] || 1) > 1
-              else
-                (s[:visits_number] || 1) > 1 &&
-                ((s[:minimum_lapse]&.floor || 1)..(s[:maximum_lapse]&.ceil || hash[:configuration][:schedule][:range_indices][:end])).none?{ |intermediate_lapse| (intermediate_lapse % 7).zero? }
-              end
-            }
-
-            raise OptimizerWrapper::DiscordantProblemError, 'Work day partition implies that lapses of all services can be a multiple of 7. There are services whose minimum and maximum lapse do not permit such lapse'
-          end
-        end
-      end
-
-      # number of visits consistency
-      if !configuration[:schedule]
-        (hash[:services].to_a + hash[:shipments].to_a).each{ |s|
-          raise OptimizerWrapper::DiscordantProblemError, 'There can not be more than one visit if no schedule is provided' unless s[:visits_number].to_i <= 1
-        }
-      end
-
-      # periodic consistency
-      return unless periodic
-
-      if hash[:relations]
-        incompatible_relation_types = hash[:relations].collect{ |r| r[:type] }.uniq - ['force_first', 'never_first', 'force_end']
-        raise OptimizerWrapper::DiscordantProblemError, "#{incompatible_relation_types} relations not available with specified first_solution_strategy" unless incompatible_relation_types.empty?
-      end
-
-      raise OptimizerWrapper::DiscordantProblemError, 'Vehicle group duration on weeks or months is not available with schedule_range_date.' if hash[:relations].to_a.any?{ |relation| relation[:type] == 'vehicle_group_duration_on_months' } &&
-                                                                                                                                                (!configuration[:schedule] || configuration[:schedule][:range_indice])
-
-      raise OptimizerWrapper::DiscordantProblemError, 'Shipments are not available with periodic heuristic.' unless hash[:shipments].to_a.empty?
-
-      raise OptimizerWrapper::DiscordantProblemError, 'Rests are not available with periodic heuristic.' unless hash[:vehicles].all?{ |vehicle| vehicle[:rests].to_a.empty? }
-
-      if hash[:configuration][:resolution][:same_point_day]
-        raise OptimizerWrapper.UnsupportedProblemError, 'Same_point_day is not supported if a set has one service with several activities' if hash[:services].any?{ |service| service[:activities].to_a.size.positive? }
-      end
+      validator = HashValidator.new(hash)
+      validator.check_hash
```

Comment:
Did you only move the validations or add some in the same time ?

Reply:
I did both, but in separated commits